src/pages/ProfileCardsPage.tsx

```diff
@@ -38,28 +38,23 @@ const createProfileFromFakerData = (): IProfile => {
   };
 };
 
-const ProfileCardsPage: FC<{ profiles: IProfile[]; setNewProfile: any }> = (
+const ProfileCardsPage: FC<{ profiles: IProfile[]; createProfile: any }> = (
   props
 ): JSX.Element => {
-  const [data, setData] = useState<IProfile[]>([]);
-
-  const handleAddProfileCard = (data: IProfile[]): void => {
-    const newData = [...data];
+  const handleAddProfileCard = (createProfile: any): void => {
     const newProfile = createProfileFromFakerData();
-    newData.push(newProfile);
-    setData(newData);
+    createProfile(newProfile);
   };
 
   useEffect(() => {
-    if (props.profiles.length <= 20) {
-      props.setNewProfile(createProfileFromFakerData());
+    if (props.profiles.length < 20) {
+      props.createProfile(createProfileFromFakerData());
     }
-    setData(props.profiles);
-  }, [props.profiles]);
+  }, [props]);
```

Comment:
removed .profiles because it was giving a dependency array warning, but would've preferred to be specific about the dependency.